apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/CurrencySelector.tsx

```diff
@@ -0,0 +1,190 @@
+import { Button } from "@/components/ui/button";
+import { FormControl } from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { CURRENCIES, type CurrencyMetadata } from "constants/currencies";
+import { useMemo, useState } from "react";
+import type { ControllerRenderProps, FieldValues } from "react-hook-form";
+import { NATIVE_TOKEN_ADDRESS, ZERO_ADDRESS, isAddress } from "thirdweb";
+import { useAllChainsData } from "../../../../../../../hooks/chains/allChains";
+
+interface CurrencySelectorProps<
+  TFieldValues extends FieldValues = FieldValues,
+> {
+  small?: boolean;
+  hideDefaultCurrencies?: boolean;
+  showCustomCurrency?: boolean;
+  isPaymentsSelector?: boolean;
+  defaultCurrencies?: CurrencyMetadata[];
+  contractChainId: number;
+  field: ControllerRenderProps<TFieldValues>;
+}
+
+export function CurrencySelector<
+  TFieldValues extends FieldValues = FieldValues,
+>({
+  small,
+  hideDefaultCurrencies,
+  showCustomCurrency = true,
+  isPaymentsSelector = false,
+  defaultCurrencies = [],
+  field,
```

Comment:
don't require passing an entire form field here - just require passing a `value` and `onChange` prop

Reply:
Does the Input field not need to be aware of the name prop that's passed into the Input?

Code for reference
```
        <Input
          {...field}
          className="w-full rounded-none"
          required
          placeholder="ERC20 Address"
          value={editCustomCurrency}
          onChange={(e) => setEditCustomCurrency(e.target.value)}
        />
 ```

Reply:
It's not a strict requirement, but you can expose the name property as well